### PR TITLE
[Kraken] terminus shchedules API: Group by StopArea

### PR DESCRIPTION
**Pull request by azime**
Terminus Shecules API: Group by StopArea.
Jira: [NAVITIAII-3145](https://jira.kisio.org/browse/NAVITIAII-3145)

_Line concerned:_
LILLE <> ROUBAIX / TOURCOING (Line:TRA:71)

image

"Gare de Lille Flandres" StopArea contains two StopPoints.
"Tourcoing Centre" StopArea contains two StopPoints.
"Roubaix Eurotéléport" StopArea contains two StopPoints.

**Case 1**: Calculate terminus_scedules at "Buisson" StopArea

Before correction:

image

After correction:

image

**Case 2**: Calculate terminus_scedules at "Gare de Lille Flandres" StopArea

Before correction:

image

After correction:

image

_Line concerned:_
MARCQ / LA MADELEINE<>HAUBOURDIN LE PARC (line:TRA:L5)

"Lille Flandres" StopArea contains two StopPoints: "Lille Flandres" and "Gare Rue De Tournai".

Direction "Le Parc" StopArea
image

Direction  "Piscine" StopArea
image

Calculate terminus_scedules at "Lille Flandres" StopArea

Before correction:

image

After correction:

image

### CODE REVIEW

**Review by kadhikari — APPROVED** [review]
1. I tested with the coverage fr-npdc customer as well as well as in local. It works fine as mentioned in the examples.
2. I also tested with the covarage fr-idfm customer as well as in local particularly with metro 13.
3. Works fine for me :+1: :+1: :+1:

File: source/time_tables/tests/departure_boards_test.cpp [hunk 1]
```diff
@@ -2390,3 +2390,352 @@ BOOST_AUTO_TEST_CASE(schedules_on_circular_routes) {
...
+    b.sa("D", 0., 0., false)("D:sp");
+
+    b.vj("Line1").route("Route1").name("VJ1")("D:sp", "08:00"_t)("C:sp", "09:00"_t)("B:sp", "10:00"_t)("Avj1",
+                                                                                                       "11:00"_t);
+    b.vj("Line1").route("Route1").name("VJ2")("C:sp", "09:10"_t)("B:sp", "10:10"_t)("Avj2", "11:10"_t);
+
+    b.finish();
+    b.data->pt_data->sort_and_index();
+    b.data->build_raptor();
+    b.data->pt_data->build_uri();
```
Comment (nonifier) [inline source/time_tables/tests/departure_boards_test.cpp]:
```suggestion
    b.make();
```
`b.make()` does all the heavy lifting for you.
Copy/pasting section of code should be a red flag for you !

File: source/time_tables/departure_boards.cpp [hunk 2]
```diff
@@ -282,17 +282,19 @@ struct JourneyPatternEndsByDirection {
 
 // For the JP of 'tested', check if all the JPP following 'tested' itself are included in the same order
 // in the JP 'direction', starting from  JPP 'direction' itself
-static bool is_jp_end_included_in_direction(const routing::JourneyPatternPoint& tested,
+static bool is_jp_end_included_in_direction(PbCreator& pb_creator,
```
Comment (nonifier) [inline source/time_tables/departure_boards.cpp]:
```suggestion
static bool is_jp_end_included_in_direction(const vector<StopPoint*>& stop_points,
```
PbCreator isn't necessary as you only need the map of stop points from `pb_creator`.
This seems harmless, but it does create unnecessary coupling and we already have waaaaaayyyyyyy to much of it already in kraken.

Comment (azime) [reply source/time_tables/departure_boards.cpp]: ok

File: source/time_tables/tests/departure_boards_test.cpp (hunk 1 above)
Comment (azime) [reply source/time_tables/tests/departure_boards_test.cpp]: ok

**Review by nonifier — APPROVED** [review]

**Pull request merged**